crypto/err/openssl.txt

```diff
@@ -2270,7 +2270,7 @@ EVP_R_NO_OPERATION_SET:149:no operation set
 EVP_R_ONLY_ONESHOT_SUPPORTED:177:only oneshot supported
 EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE:150:\
 	operation not supported for this keytype
-EVP_R_OPERATON_NOT_INITIALIZED:151:operaton not initialized
+EVP_R_OPERATON_NOT_INITIALIZED:151:operation not initialized
```

Comment:
This would be better fixed by correcting the reason code and regenerating.

Reply:
> This would be better fixed by correcting the reason code and regenerating.

Since this changes a symbolic constant, and since we are targeting 1.1.1, wouldn't that require a compatibility define like the following somewhere? (cc @levitte)
```C
#define EVP_R_OPERATON_NOT_INITIALIZED  EVP_R_OPERATION_NOT_INITIALIZED
```

Reply:
@paulidale, I don't know what is "the reason code" here. But we can cherry pick everything else and ignore that single line change in order to address it in a separate pull request if you want.

@mspncp I believe I deliberately avoided renaming public methods/constants to make it easy to merge. Adding a `EVP_R_OPERATION_NOT_INITIALIZED` could be done in a separate PR.

Reply:
> @paulidale, I don't know what is "the reason code" here.

@Coeur sorry for talking 'over your head' with @paulidale: the "reason code" is the symbolic constant (here `EVP_R_OPERATON_NOT_INITIALIZED`) from which the  error message ('operaton not initialized') is generated automagically by some perl script. That's why the correct way to fix the error message would be to fix the name of the constant. But that is problematic, because applications might check the error constant, and changing its name could brake application code.

So your suggestion makes perfect sense for me: I'll just drop the one line from openssl.txt when merging your commit. Then @paulidale and I can still decide whether it is worth to fix it in a separate pull request. (Probably not.)

Reply:
This "reason code" was not fixed in master too, just its message.
See https://github.com/openssl/openssl/blob/0c789f59f117ccbb30ffc621216ba776117c7c61/crypto/err/openssl.txt#L2478
I just also grep this typo in BoringSSL ;)

Reply:
Then we should at least fix it properly in master. But also on master the question (which goes to @openssl/omc) remains:  how to properly deal with typo corrections in error constants or similar preprocessor defines? Do we need a compatibility define like
```C
#define EVP_R_OPERATON_NOT_INITIALIZED  EVP_R_OPERATION_NOT_INITIALIZED
```
as mentioned in https://github.com/openssl/openssl/pull/9295#discussion_r309127192, guarded by deprecation guard?

Reply:
To my astonishment, there are other errors where the name of the reason code and the text differ. Just one line below:

https://github.com/openssl/openssl/blob/8e7b35e9110f89a81de88f9a1fbc25b5db1eec0d/crypto/err/openssl.txt#L2273-L2274

I thought this was enforced by the perl script. But apparently this holds only for new errors (unless a full regeneration of the error codes is enforced). Well, then I might as well could have just kept the typo correction of the error text. But now it's too late, sigh.

Reply:
Aren't the errors regenerated from time to time?
Would this result in messages being reverted?